packages/tpetra/core/src/Tpetra_Details_unpackCrsMatrixAndCombine_def.hpp

```diff
@@ -284,7 +284,7 @@ struct UnpackCrsMatrixAndCombineFunctor {
 
     if (expected_num_bytes > num_bytes)
     {
-      printf(
+      KOKKOS_IMPL_DO_NOT_USE_PRINTF(
```

Comment:
I prefer not to use KOKKOS_IMPL details in Tpetra.  Is there a problem with printf and SYCL?  What does KOKKOS_IMPL_DO_NOT_USE_PRINTF do?  How would one know to use this macro rather than printf?

Reply:
Oh, I should have pointed this out directly. `SYCL` doesn't support `printf` on the device. In `Kokkos` we are defining a workaround to make this work regardless with the intention of only using it internally (such that it could be removed at any point). The alternative is to just guard these places with `#ifndef KOKKOS_ENABLE_SYCL` or similar.

Reply:
yes there is a problem with printf in SYCL. Namely that it isn't supported currently and you have to call sycl::printf or something like that ... 
We are talking to various compiler folks to get that changed, and thus didn't want to add an official portability functionality like KOKKOS_PRINTF yet. But without changing it this guy won't compile/work with SYCL right now.

Reply:
Probably doing #ifndef KOKKOS_ENABLE_SYCL around the printfs is better than using a Kokkos implementation detail.  Plus, it documents why we are doing something different from printf -- that is, that SYCL alone doesn't like it.  When Kokkos adds an official portable KOKKOS_PRINTF, we will adopt it.

Reply:
Fixed in [`45fefc7` (#9086)](https://github.com/trilinos/Trilinos/pull/9086/commits/45fefc7f17a789cc870b1b1e1525157ccf6126df).